Jenkinsfile

```diff
@@ -831,17 +831,79 @@ pipeline {
+                            sh '''rm -rf src/tests/ftest/avocado/job-results/*/html/
+                                  if [ -n "$STAGE_NAME" ]; then
+                                      rm -rf "$STAGE_NAME/"
+                                      mkdir "$STAGE_NAME/"
+                                      ls *daos{,_agent}.log* >/dev/null && mv *daos{,_agent}.log* "$STAGE_NAME/"
+                                      mv src/tests/ftest/avocado/job-results/* \
+                                         $(ls src/tests/ftest/*.stacktrace || true) "$STAGE_NAME/"
+                                  fi'''
+                            junit env.STAGE_NAME + '/*/results.xml, src/tests/ftest/*_results.xml'
+                            archiveArtifacts artifacts: env.STAGE_NAME + '/**'
+                        }
+                        /* temporarily moved into runTest->stepResult due to JENKINS-39203
+                        success {
+                            githubNotify credentialsId: 'daos-jenkins-commit-status',
+                                         description: env.STAGE_NAME,
+                                         context: 'test/' + env.STAGE_NAME,
+                                         status: 'SUCCESS'
+                        }
+                        unstable {
+                            githubNotify credentialsId: 'daos-jenkins-commit-status',
+                                         description: env.STAGE_NAME,
+                                         context: 'test/' + env.STAGE_NAME,
+                                         status: 'FAILURE'
+                        }
+                        failure {
+                            githubNotify credentialsId: 'daos-jenkins-commit-status',
+                                         description: env.STAGE_NAME,
+                                         context: 'test/' + env.STAGE_NAME,
+                                         status: 'ERROR'
+                        }
+                        */
+                    }
+                }
+                stage('Functional_Hardware') {
+                    agent {
+                        label 'ci_nvme9'
+                    }
+                    steps {
+                        // First snapshot provision the VM at beginning of list
+                        provisionNodes NODELIST: env.NODELIST,
+                                       node_count: 1,
+                                       snapshot: true
+                        // Then just reboot the physical nodes
+                        provisionNodes NODELIST: env.NODELIST,
+                                       node_count: 9,
+                                       power_only: true
+                        runTest stashes: [ 'CentOS-install', 'CentOS-build-vars' ],
+                                script: '''export PDSH_SSH_ARGS_APPEND="-i ci_key"
+                                           test_tag=$(git show -s --format=%B | sed -ne "/^Test-tag:/s/^.*: *//p")
```

Comment:
Should we support different `Test-tag:` pragmas now that we have different/multiple Functional test stages?

Reply:
That would be up to a future patch.  This is just adding the hardware CI stage.

Reply:
Except that adding the hardware stage is changing the meaning of the test tags and now there are separate sets of them and using `Test-tag:` is going to set the tags the same for both hardware and VM testing, even though by default they are different.

IOW, it seems to me that now that there are multiple Functional stages, there should be multiple pragmas to set their respective tags, particularly given that by default they are not even the same.